components/question/Rating/index.jsx

```diff
@@ -0,0 +1,58 @@
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import { useEffect, useState } from 'react';
+
+import { buildSurveyQuestionsWithAnswers } from '../Details/buildSurveyQuestionsWithAnswers';
+import { emojiMapper } from './emojiMapper';
+
+export const ratingTestIds = {
+  base: 'rating',
+  emoji: 'rating__emoji',
+};
+
+const Rating = ({ id, type, answers, setSurveyQuestionsWithAnswers }) => {
+  const [currentRating, setCurrentRating] = useState(-1);
+  const emoji = emojiMapper[type];
+  const totalEmojis = [...Array(answers.length)];
+
+  useEffect(() => {
+    setCurrentRating(-1);
+  }, [id]);
+
+  const onRatingChange = index => {
+    const surveyQuestionsWithAnswers = buildSurveyQuestionsWithAnswers(id, answers[index].id);
+    setCurrentRating(index);
+    setSurveyQuestionsWithAnswers(prevSurveyQuestionWithAnswer => {
+      let newSurveyQuestionWithAnswer = prevSurveyQuestionWithAnswer.slice();
+      newSurveyQuestionWithAnswer = newSurveyQuestionWithAnswer.filter(response => response.id !== id);
+
+      return [...newSurveyQuestionWithAnswer, surveyQuestionsWithAnswers];
+    });
+  };
+
+  return (
+    <div className="w-[215px] m-auto mt-8" data-test-id={ratingTestIds.base}>
+      <ul className="flex">
+        {totalEmojis.map((_key, index) => (
+          <li
+            className={classNames('cursor-pointer text-base-xxxl tracking-[16px]', { 'opacity-50': index > currentRating })}
```

Comment:
Not sure whether we should use `width` instead of `letter-spacing` or not as currently, it seems like the icon is not quite centered 💭 

![Screen Shot 2565-05-18 at 16 29 00](https://user-images.githubusercontent.com/1772999/169008462-94d5170d-b39a-412f-9051-ce2190cb3100.png)

Reply:
sure, applied `justify-between` on the parent flex container, and removed the `letter-spacing` too in 8c75f0e6894c591a2ba4bcb83910a5fff521f136, Thanks!